bpfman-operator/apis/v1alpha1/cluster_fentry_program_types.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2024 The bpfman Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+// ClFentryProgramInfo defines the Fentry program details
+type ClFentryProgramInfo struct {
+	ClFentryLoadInfo `json:",inline"`
+	// Whether the program should be attached to the function.
+	// +optional
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:default:={}
+	Links []ClFentryAttachInfo `json:"links"`
+}
+
+// ClFentryLoadInfo contains the program-specific load information for Fentry
+// programs
+type ClFentryLoadInfo struct {
+	// function is the name of the function to attach the Fentry program to.
+	Function string `json:"function"`
+}
+
+type AttachType string
+
+const (
+	Attach  AttachType = "true"
+	Dettach AttachType = "false"
```

Comment:
A couple things here:
- We typically recommend a name format of `{alias}{constValue}`. So `Attach` would be `AttachTypeAttach`.
- These constant values don't match the values that a user is allowed to specify where this is used. A user can only input the string literals `Attach` and `Detach`, not `true` and `false`. You should have these values match what users can input so you can accurately do comparisons in code with these constants.